mtags/src/main/scala-2/scala/meta/internal/pc/MetalsGlobal.scala

```diff
@@ -178,6 +178,35 @@ class MetalsGlobal(
       this(string + ".", string)
   }
 
+  private def backtickify(
+      tpe: Type,
+      loop: (Type, Option[ShortName]) => Type,
+      withPrefix: Boolean
+  ): Type =
+    tpe match {
+      case TypeRef(pre, sym, args)
+          if Identifier.needsBacktick(sym.decodedName) &&
+            sym != definitions.ByNameParamClass && // `=> T` is OK
+            !sym.isPackageObject &&
+            !definitions.isRepeated(sym) &&
+            !sym.isAbstractType &&
```

Comment:
It seems that this approach isn't reliable.
`sym.decodedName` is different from the string representation of type.

Instead of checking the whole value we need to check each part of type. But I understand that it might be complicated.

Maybe as alternative to this, we can use scalameta parser?
I mean, we can turn type into string but instead passing it directly into textEdit use scalameta parser and then reconstruct it and applying backticks?